src/components/ShoppingCart/dropDownButton.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useState } from "react";
+import { GrDown, GrUp } from "react-icons/gr";
+import { IconButton } from "@chakra-ui/react";
+import { Collapse } from "@chakra-ui/react";
+import { PriceBar } from "@/store/reducer/priceBarSlice";
+
+type detail = {
+  detail: PriceBar[];
+};
+
+const DropDownButton = ({ detail }: detail) => {
+  const [show, setShow] = useState(false);
+
+  const handleToggle = () => setShow(!show);
+
+  const scroll = "scroll";
+  const hidden = "hidden";
+  const noWrap = "nowrap";
+  const normal = "normal";
+  const clip = "clip";
+  const ellipsis = "ellipsis";
+
+  return (
+    <div
+      style={{
+        width: "100%",
+        display: "flex",
+        flexDirection: "row",
+        justifyContent: "center",
+        alignContent: "center",
+        overflow: "auto",
+      }}
+    >
+      <Collapse
+        startingHeight={25}
+        in={show}
+        style={{
+          width: "100%",
+          overflowY: show ? scroll : hidden,
+          userSelect: "none",
+          whiteSpace: show ? normal : noWrap,
+          textOverflow: show ? clip : ellipsis,
+          textAlign: "center",
+        }}
+        data-testid="testShow"
+      >
+        {detail.map(
+          (content: PriceBar) => `Color:${content.color},  Quantity:${content.quantity},  `
+        )}
```

Comment:
The detailed information can be put into a Box container and added with some necessary props instead of writing inline style in the Collapse component.

Reply:
I have tried to put the content in Box. But some function will lose if I do that. In order to avoid too much inline style, I put all style property into const. 


 const collapseStyle: MotionStyle = {
          width: "100%",
          overflowY: show ? "scroll" : "hidden",
          userSelect: "none",
          whiteSpace: show ? "normal" : "nowrap",
          textOverflow: show ? "clip" : "ellipsis",
          textAlign: "center"
        }

 <Collapse
        startingHeight={25}
        in={show}
        style={collapseStyle}
        data-testid="testShow"
      >

It should be better now.